### ISSUE TITLE
Warn if files in Pages of @content, @index do not exist

### ISSUE THREAD

**Issue by goerz** [JuliaDocs/Documenter.jl#2302]
While writing the test for https://github.com/JuliaDocs/DocumenterCitations.jl/pull/48/, I stumbled on some issues with the `Pages` attribute in `@content` and `@index` blocks:

1. There is no warning if a file listed in `Pages` does not exist. I'd even argue that in strict mode (default), referencing a file that does not exist should throw an exception.
2. `Pages` in `@content` and `@index` blocks does not have `normpath` applied to it, which is inconsistent with `@autodocs`, as already noted in https://github.com/JuliaDocs/DocumenterCitations.jl/issues/22#issuecomment-1760195160. This is especially pernicious in combination with (1), as people are probably somewhat likely to just write unix path separators (`/`), and then have their documentation build be subtly wrong (but not fail) on Windows.

Arguably, `@autodocs` should also warn if its `Pages` contains an entry that doesn't match anything.

Both of the above problems are handled in DocumenterCitation's `@bibliography` block, leading to the inconsistencies shown in this example:

https://github.com/JuliaDocs/DocumenterCitations.jl/pull/48/files#diff-94253cd814e34a8eed4ea72a3fe7c963d73ea2149f6f080d42a8f933545c8bf1

**Comment by mortenpi** [comment]
> I'd even argue that in strict mode (default), referencing a file that does not exist should throw an exception.

I agree that this would be ideal. I am wondering how we should handle new build-breaking errors now that we have promised not to break things for a while (i.e. whether this would constitute a breaking change or not).

**Comment by laborg** [comment]
As a user, I would warmly accept this “breaking” change as it would help me avoid providing incorrect documentation (missing pages!).

**Comment by goerz** [comment]
Personally, I wouldn’t think of this as “breaking”, but I suppose it’s debatable.
